src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -0,0 +1,358 @@
+                }
+                while (!Unsafe.IsAddressGreaterThan(ref currentRightSearchSpace, ref oneVectorAwayFromRightEnd));
+
+                // If any elements remain, process the last vector in the search space.
+                if ((uint)right.Length % Vector128<T>.Count != 0)
+                {
+                    rightValues = Vector128.LoadUnsafe(ref oneVectorAwayFromRightEnd);
+                    leftValues = Vector128.LoadUnsafe(ref oneVectorAwayFromLeftEnd);
+
+                    if (Vector128.Equals(leftValues, rightValues) != Vector128<T>.AllBitsSet)
+                    {
+                        return false;
+                    }
+
+                    if (VectorContainsAnyNonAsciiData(leftValues) || VectorContainsAnyNonAsciiData(rightValues))
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether the provided buffers contain equal ASCII characters, ignoring case considerations.
+        /// </summary>
+        /// <param name="left">The buffer to compare with <paramref name="right" />.</param>
+        /// <param name="right">The buffer to compare with <paramref name="left" />.</param>
+        /// <returns><see langword="true" /> if the corresponding elements in <paramref name="left" /> and <paramref name="right" /> were equal ignoring case considerations and ASCII. <see langword="false" /> otherwise.</returns>
+        /// <remarks>If both buffers contain equal, but non-ASCII characters, the method returns <see langword="false" />.</remarks>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<byte> right)
+            => left.Length == right.Length && SequenceEqualIgnoreCase(left, right);
+
+        /// <inheritdoc cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
+            => left.Length == right.Length && SequenceEqualIgnoreCase(right, left);
+
+        /// <inheritdoc cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<char> left, ReadOnlySpan<char> right)
+            => left.Length == right.Length && SequenceEqualIgnoreCase(right, left);
+
+        private static bool SequenceEqualIgnoreCase<TLeft, TRight>(ReadOnlySpan<TLeft> left, ReadOnlySpan<TRight> right)
+            where TLeft : unmanaged, INumberBase<TLeft>
+            where TRight : unmanaged, INumberBase<TRight>
+        {
+            Debug.Assert(left.Length == right.Length);
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                uint valueA = uint.CreateTruncating(left[i]);
+                uint valueB = uint.CreateTruncating(right[i]);
+
+                if (!UnicodeUtility.IsAsciiCodePoint(valueA) || !UnicodeUtility.IsAsciiCodePoint(valueB))
```

Comment:
```suggestion
                if (!UnicodeUtility.IsAsciiCodePoint(valueA | valueB))
```
Should work too and saves a branch. The MSB will only be set if one of the values is > 0x7F (non-ASCII).